### PR TITLE
[hive]add retention seconds in hive catalog.

### PR DESCRIPTION
**Pull request by YanZhangN**
### Purpose
Add `retention` properties for hive metastore according to 'partition.expiration-time'.

Linked issue: close #xxx

### CODE REVIEW

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 1]
```diff
@@ -814,6 +814,11 @@ private Table newHmsTable(
         if (provider == null) {
             provider = "paimon";
         }
+        CoreOptions coreOptions = CoreOptions.fromMap(tableParameters);
+        int retention =
+                coreOptions.partitionExpireTime() == null
+                        ? Integer.MAX_VALUE
+                        : (int) coreOptions.partitionExpireTime().getSeconds();
```
[the remark this answers is not in the document]
Comment (YanZhangN) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java]: fixed

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 2]
```diff
@@ -814,6 +814,11 @@ private Table newHmsTable(
         if (provider == null) {
             provider = "paimon";
         }
+        CoreOptions coreOptions = CoreOptions.fromMap(tableParameters);
```
[the remark this answers is not in the document]
Comment (YanZhangN) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:817]: fixed

**Comment by JingsongLi** [comment]
Can you provide the specific application of this attribute? For example, Hive first introduced it with JIRA, etc.

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 3]
```diff
@@ -814,6 +814,12 @@ private Table newHmsTable(
...
+        java.time.Duration partitionExpireTime = coreOptions.partitionExpireTime();
+        int retention = Integer.MAX_VALUE;
+        if (partitionExpireTime != null && partitionExpireTime.getSeconds() <= Integer.MAX_VALUE) {
+            retention = (int) partitionExpireTime.getSeconds();
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:821]: Maybe only set it with metastore.partitioned-table enabled.

**Comment by JingsongLi** [comment]
I am not sure this field is for partition retention.

Maybe we can use properties to store this option.

**Pull request closed without merging**